lib/api/route/typing.dart

```diff
@@ -0,0 +1,26 @@
+import '../core.dart';
+import '../model/events.dart';
+import 'messages.dart';
+
+
+/// https://zulip.com/api/set-typing-status
+Future<void> setTypingStatus(ApiConnection connection, {
+  required TypingOp op,
+  required MessageDestination destination,
+}) {
+  final supportsTypeChannel = connection.zulipFeatureLevel! >= 248; // TODO(server-9)
+  switch (destination) {
+    case StreamDestination():
+      return connection.post('setTypingStatus', (_) {}, 'typing', {
+        'type': RawParameter((supportsTypeChannel) ? 'channel' : 'stream'),
+        'op': RawParameter(op.toJson()),
+        'stream_id': destination.streamId,
```

Comment:
The API doc says 'stream_id' is new in FL 215, and there's a different way we need to express the destination for <215.

Reply:
Interesting, thanks for catching that!